Automattic-Tracks-iOS/Crash Logging/SentryEvent+Extensions.swift

```diff
@@ -0,0 +1,33 @@
+import Sentry
+
+extension Event {
+    static func from(
+        error: NSError,
+        level: SentryLevel = .error,
+        user: Sentry.User? = nil,
+        timestamp: Date = Date(),
+        extra: [String: Any] = [:]
+    ) -> Event {
+        let event = Event(level: level)
+
+        if let underlyingError = error.userInfo[NSUnderlyingErrorKey] as? NSError {
+            event.message = SentryMessage(formatted: underlyingError.debugDescription)
+        } else {
+            event.message = SentryMessage(formatted: error.debugDescription)
+        }
+
+        /// The error code/domain can be used to reproduce the original error
+        var mutableExtra = extra
+        mutableExtra["error-code"] = error.code
+        mutableExtra["error-domain"] = error.domain
+
+        error.userInfo.forEach { (key, value) in
+            mutableExtra[key] = value
+        }
```

Comment:
I'm wondering if that works in all cases – I mean with the fact that `mutableExtra` is `[String: Any]` I wonder how Sentry serialize that extra dictionary when the `Any` values are not JSON-compatible types?

For example, if your `error` has an `userInfo[NSUnderlyingErrorKey]`, whose value is typically itself another `NSError`, that means that `mutableExtra` will end up with one of its value being an `NSError`, and I wonder how that will be serialized and show up in Sentry?

Reply:
Sentry doesn't use `Codable` for JSON, so it'll just print the object description for anything that's invalid. Adding 

```swift
        mutableExtra["foo"] = 1 as Any
        mutableExtra["bar"] = DispatchQueue(label: "test")
```

to the list results in this:

<img width="949" alt="Screen Shot 2021-03-24 at 10 34 20 PM" src="https://user-images.githubusercontent.com/1123407/112419517-2d4d9c00-8cf1-11eb-8137-271589144b26.png">

which is fine to me. 

The reason I don't mind keeping the entire `userInfo[NSUnderlyingErrorKey]` is that we don't extract all the data from it, so whatever else it can offer us may be useful for debugging – IMHO it doesn't hurt to have it?

Reply:
Yep, looks good 👍  As long as Sentry doesn't crash and instead properly use the object `description` in those cases, that's all good to me 👍